src/main/java/org/eclipse/keyple/core/service/SmartCardService.java

```diff
@@ -0,0 +1,192 @@
+        throw new IllegalStateException(
+            "Plugin has already been registered to the platform : " + pluginName);
+      } else {
+        Plugin pluginInstance = pluginFactory.getPlugin();
+        if (pluginInstance instanceof AbstractPlugin) {
+          logger.info("Registering a new Plugin to the platform : {}", pluginName);
+          ((AbstractPlugin) pluginInstance).register();
+        } else {
+          logger.info("No registration needed for pool plugin : {}", pluginName);
+        }
+        this.plugins.put(pluginName, pluginInstance);
+        return pluginInstance;
+      }
+    }
+     */
+    return null;
+  }
+
+  /**
+   * Unregister plugin from platform
+   *
+   * @param pluginName plugin name
+   * @throws IllegalStateException if the plugin or his reader(s) are already unregistered
+   */
+  public void unregisterPlugin(String pluginName) {
+    /*
+    synchronized (MONITOR) {
+      final Plugin removedPlugin = plugins.remove(pluginName);
+      if (removedPlugin != null) {
+        if (removedPlugin instanceof AbstractPlugin) {
+          ((AbstractPlugin) removedPlugin).unregister();
+          logger.info("Unregistering a plugin from the platform : {}", removedPlugin.getName());
+        } else {
+          logger.info("Unregistration not needed for pool plugin : {}", pluginName);
+        }
+      } else {
+        throw new IllegalStateException(
+            String.format("This plugin, %s, is not registered", pluginName));
+      }
+    }
+     */
+  }
+
+  /**
+   * Check weither a plugin is already registered to the platform or not
+   *
+   * @param pluginName name of the plugin to be checked
+   * @return true if a plugin with matching name has been registered
+   */
+  public synchronized boolean isRegistered(String pluginName) {
+    synchronized (MONITOR) {
+      return plugins.containsKey(pluginName);
+    }
+  }
+
+  /**
+   * Gets the plugins.
+   *
+   * @return the plugin names and plugin instances map of interfaced reader’s plugins.
+   */
+  public synchronized Map<String, Plugin> getPlugins() {
+    return plugins;
+  }
+
+  /**
+   * Gets the plugin whose name is provided as an argument.
+   *
+   * @param name the plugin name
+   * @return the plugin
+   * @throws KeyplePluginNotFoundException if the wanted plugin is not found
+   */
+  public synchronized Plugin getPlugin(String name) {
+    synchronized (MONITOR) {
+      Plugin plugin = plugins.get(name);
+      if (plugin == null) {
+        throw new KeyplePluginNotFoundException(name);
+      }
+      return plugin;
+    }
+  }
+
+  /**
+   * Verifies the compatibility with the service of the provided card extension.
+   *
+   * <p>The verification is based on the comparison of the respective API versions.
+   *
+   * @param cardExtension A not null {@link KeypleCardExtension} reference object
+   */
+  public void checkCardExtension(KeypleCardExtension cardExtension) {
+    // TODO complete
+  }
+
+  /**
+   * TODO complete
```

Comment:
Commentaire : Registers a new Distributed Local Service to be available in the platform if not registered yet